### PR TITLE
Add pre up option

### PR DESCRIPTION
**Pull request by smutel**
Hello,

* Add ansible manage comment on top of each template
* Add an option to setup pre-up command
* Avoid multiple spaces in network interfaces configuration file

### CODE REVIEW

**Comment by michaelrigart** [comment]
@smutel : thank you for the PR. Is it possible to change the pre-up variable to use an underscore instead of a dash and perhaps document it in the readme?

@markgoddard : I don't see any issues adding the option. Any other remarks?

**Comment by markgoddard** [comment]
@michaelrigart, I'll take a look shortly.

**Comment by smutel** [comment]
I did the requested changes.  
I put something in the README, however it does not contain array with a list of variables so I don't know if I put the description in the good place.

File: templates/ethernet_Debian.j2 [hunk 1]
```diff
@@ -1,4 +1,9 @@
+# {{ ansible_managed }}
+
 {% if item.bootproto == 'static' %}
+{% if item.pre_up is defined %}
+pre-up {{ item.pre_up }}
+{% endif %}
```
Comment (markgoddard) [inline templates/ethernet_Debian.j2:6]: Placing this here means if bootproto is DHCP then pre-up won't work. You could move before the first 'if', or duplicate on L32.

Also, does it make sense to also support this for bonds and bridges?

**Review by markgoddard — COMMENTED** [review]
Ideally we wouldn't change whitespace to avoid unnecessary restarts of interfaces, but since the config files are changing anyway I suppose it's ok.

File: templates/ethernet_Debian.j2 (hunk 1 above)
Comment (smutel) [reply templates/ethernet_Debian.j2:6]: I put the option before the ip/proto.

Regarding the man, this option is explained in the iface options with this message `The  following  "command" options are available for every family and method.`
In this section we have pre-up, up, post-up, down, pre-down, post-down and description.

Tell me if you want me to add more options, in other files...

Comment (markgoddard) [reply templates/ethernet_Debian.j2:6]: I think at least it should be consistent for ethernet, bridge and bond interfaces. If you want to add the other command options, feel free.

File: templates/bond_Debian.j2 [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# {{ ansible_managed }}
+
+{% if item.pre_up is defined %}
+pre-up {{ item.pre_up }}
```
Comment (markgoddard) [inline templates/bond_Debian.j2:4]: endif

File: templates/ethernet_Debian.j2 [hunk 3]
```diff
@@ -1,4 +1,9 @@
+# {{ ansible_managed }}
+
+{% if item.pre_up is defined %}
+pre-up {{ item.pre_up }}
```
Comment (markgoddard) [inline templates/ethernet_Debian.j2:4]: move endif to here

File: templates/bridge_Debian.j2 [hunk 4]
```diff
@@ -1,3 +1,7 @@
+# {{ ansible_managed }}
+
+{% if item.pre_up is defined %}
+pre-up {{ item.pre_up }}
```
Comment (markgoddard) [inline templates/bridge_Debian.j2:4]: endif

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: templates/bond_Debian.j2 (hunk 2 above)
Comment (smutel) [reply templates/bond_Debian.j2:4]: Sorry. Done.

File: templates/ethernet_Debian.j2 (hunk 3 above)
Comment (smutel) [reply templates/ethernet_Debian.j2:4]: Sorry. Done.

File: templates/bridge_Debian.j2 (hunk 4 above)
Comment (smutel) [reply templates/bridge_Debian.j2:4]: Sorry. Done.

**Review by markgoddard — APPROVED** [review]
Thanks

**Comment by markgoddard** [comment]
Released as v1.10.0

**Pull request merged**